### PR TITLE
bug 1185751 - Get ADI pulled in on stage

### PR DESCRIPTION
**Pull request by peterbe**
@twobraids r?

I did some indentation fixes here and there too. That's why it looks like a bigger diff. 
Those changes makes the code read better I think. 

I'm now going to try to set up a crontabber instance locally that just runs this job.

### CODE REVIEW

File: socorro/cron/jobs/fetch_adi_from_hive.py [hunk 1]
```diff
@@ -183,13 +176,99 @@ class FetchADIFromHiveCronApp(BaseCronApp):
...
+        tx_class = self.config.primary_destination.transaction_executor_class
+        primary_transaction = tx_class(
+            self.config,
+            primary_database,
+        )
+        transactions = [primary_transaction]
+
+        db_class = self.config.secondary_destination.database_class
+        secondary_database = db_class(self.config.secondary_destination)
+        if secondary_database.config != primary_database.config:
```
Comment (twobraids) [inline socorro/cron/jobs/fetch_adi_from_hive.py]: nice touch

**Comment by peterbe** [comment]
How I tested this locally:

1. Make sure you're on VPN
2. Generate a new crontabber.ini file with `python socorro/cron/crontabber_app.py --admin.dump_conf=../crontabber-hive.ini`
3. Edit `../crontabber-hive.ini` and make sure the `jobs=` line looks like this:
    ```
    jobs=socorro.cron.jobs.fetch_adi_from_hive.FetchADIFromHiveCronApp|1d
    ```
4. Create a new database and load in the same stuff you have in your local `breakpad` table. 
    ```
    createdb breakpad_deleteme
    pg_dump breakpad > /tmp/breakpad.sql
    psql breakpad_deleteme < /tmp/breakpad.sql
    ```
5. Remove (or ignore) the config of all other jobs and focus on FetchADIFromHiveCronApp. Here's what mine looked like:
    ```
   [[class-FetchADIFromHiveCronApp]]
        # Hostname to run Hive query on
        hive_host={{REDACTED IN CASE IT MATTERS}}
        [[[secondary_destination]]]

            # the name of the database
            # see "resource.postgresql.database_name" for the default or override it here
            database_name=breakpad_deleteme
    ```
Note that I had to manually delete the `#query=` part. The only overriding changes should be `hive_host` and `secondary_destination.database_name`. 
6. Run it `python socorro/cron/crontabber_app.py --admin.conf=../crontabber-hive.ini`
7. Wait. About 10min
8. `psql breakpad` and `select count(*) from raw_adi_logs`
9. `psql breakpad_deleteme` and `select count(*) from raw_adi_logs`

Should have about 466k records in each.

File: socorro/cron/jobs/fetch_adi_from_hive.py [hunk 2]
```diff
@@ -183,13 +176,99 @@ class FetchADIFromHiveCronApp(BaseCronApp):
...
+        db_class = self.config.primary_destination.database_class
+        primary_database = db_class(self.config.primary_destination)
+        tx_class = self.config.primary_destination.transaction_executor_class
+        primary_transaction = tx_class(
+            self.config,
+            primary_database,
+        )
+        transactions = [primary_transaction]
+
+        db_class = self.config.secondary_destination.database_class
```
Comment (twobraids) [inline socorro/cron/jobs/fetch_adi_from_hive.py]: actually, perhaps these four lines would be better as:
```python
    if  (
        db_class = self.config.secondary_destination.database_class
        and self.config.secondary_destination != self.config.primary_destination
    ):
        db_class = self.config.secondary_destination.database_class
        secondary_database = db_class(self.config.secondary_destination)
```
That way we can turn off the secondary by setting the class to the empty string.

Comment (peterbe) [inline socorro/cron/jobs/fetch_adi_from_hive.py]: You mean to change 
```python
    required_config.secondary_destination.add_option(
        'database_class',
        default=ConnectionContext,
        doc='the class responsible for connecting to Postgres',
        reference_value_from='resource.postgresql',
    )
```
...to...:
```python
    required_config.secondary_destination.add_option(
        'database_class',
        default='',
        doc='the class responsible for connecting to Postgres',
        reference_value_from='resource.postgresql',
    )
```
I'm not convinced. Its confusing what the value should be. And it's something else you also need to configure if you want to enable the second destination. Right now you just need to override the database credentials. Having to also configure the `database_class` would be another thing you have to get right.

**Comment by twobraids** [comment]
r+ pending changes from conversation

**Pull request merged**